core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala

```diff
@@ -276,6 +289,173 @@ class ConsumerBounceTest extends BaseRequestTest with Logging {
     future2.get
   }
 
+  /**
+    * If we have a running consumer group of size N, configure consumer.group.max.size=N-1 and restart all brokers,
+    * the group should be forced to rebalance when it becomes hosted on a Coordinator with the new config.
+    * Then, 1 consumer should be left out of the group
+    */
+  @Test
+  def testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup(): Unit = {
```

Comment:
This test fails for me about 50% of the time when I run it locally. I wonder if you see this flakiness on your side.
Sample error messages:
```
java.lang.AssertionError: expected:<102> but was:<105>

org.scalatest.junit.JUnitTestFailedError: Expected to only receive one exception of type
class org.apache.kafka.common.errors.GroupMaxSizeReachedExceptionduring consumption.
Received: ArrayBuffer(org.apache.kafka.clients.consumer.CommitFailedException: Commit
cannot be completed since the group has already rebalanced and assigned the partitions to
another member. This means that the time between subsequent calls to poll() was longer
than the configured max.poll.interval.ms, which typically implies that the poll loop is
spending too much time message processing. You can address this either by increasing
max.poll.interval.ms or by reducing the maximum size of batches returned in poll() with
max.poll.records.)

java.lang.AssertionError: expected:<102> but was:<136>
```

Reply:
Not 50% of the time but I do see some flakiness now. What do you think would be a good way to go around this? Checking for at least X records rather than exact?

Reply:
After the changes, this test has passed 50/50 times for me

Reply:
Yes, it now passes for me too (20/20). Thanks!